packages/protocol/test-sol/stability/PairManager.t.sol

```diff
@@ -0,0 +1,431 @@
+
+  function test_initilize_shouldSetOwner() public view {
+    assert(testee.owner() == deployer);
+  }
+
+  function test_initilize_shouldSetBroker() public view {
+    assert(testee.broker() == broker);
+  }
+
+  /* ---------- Setters ---------- */
+
+  function test_setBroker_whenSenderIsNotOwner_shouldRevert() public {
+    changePrank(notDeployer);
+    vm.expectRevert("Ownable: caller is not the owner");
+    testee.setBroker(address(0));
+  }
+
+  function test_setBroker_whenAddressIsZero_shouldRevert() public {
+    vm.expectRevert("Broker address must be set");
+    testee.setBroker(address(0));
+  }
+
+  function test_setBroker_whenSenderIsOwner_shouldUpdateAndEmit() public {
+    address newBroker = actor("newBroker");
+    vm.expectEmit(true, false, false, false);
+    emit BrokerUpdated(newBroker);
+
+    testee.setBroker(newBroker);
+
+    assert(testee.broker() == newBroker);
+  }
+
+  function test_updateBuckets_whenSenderIsNotBroker_shouldRevert() public {
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        testStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    vm.expectRevert("Caller is not the Broker");
+    testee.updateBuckets(pairId, 45000000000, 45000000000);
+  }
+
+  function test_updateBuckets_whenPairDoesNotExist_shouldRevert() public {
+    changePrank(broker);
+
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        anotherTestStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    vm.expectRevert("A pair with the specified id does not exist");
+    testee.updateBuckets(pairId, 45000000000, 45000000000);
+  }
+
+  function test_updateBuckets_whenSenderIsBroker_shouldUpdateAndEmit() public {
+    changePrank(broker);
+
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        testStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    PairManager.Pair memory pairBeforeUpdate = testee.getPair(pairId);
+    assert(pairBeforeUpdate.stableBucket == 250000000000);
+    assert(pairBeforeUpdate.collateralBucket == 250000000000);
+
+    vm.expectEmit(false, false, false, false);
+    emit BucketsUpdated(pairId, 45000000000, 45000000000);
+    testee.updateBuckets(pairId, 45000000000, 45000000000);
+
+    PairManager.Pair memory pairAfterUpdate = testee.getPair(pairId);
+    assert(pairAfterUpdate.stableBucket == 45000000000);
+    assert(pairAfterUpdate.collateralBucket == 45000000000);
+  }
+
+  /* ---------- Getters ---------- */
+
+  function testFail_getPair_whenPairDoesNotExist_shouldRevert() public {
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        anotherTestStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    // vm.expectRevert("A pair with the specified id does not exist");
```

Comment:
What's up here? I noticed this as well.

Reply:
So when i use this to verify the revert, the test fails even though the tx reverts with the expected message.
![image](https://user-images.githubusercontent.com/6872903/187571183-b7cca973-094a-4565-a00a-f539d0740323.png)

So I removed expect revert and used testFail instead. I have no idea why the expect was failing...